### ISSUE TITLE
Cannot invoke Amazon Nova model with llm-bedrock plugin

### ISSUE THREAD

**Issue by indrayam** [simonw/llm-bedrock#14]
### Issue Summary
Amazon Nova models cannot be invoked using the `llm bedrock` plugin

### My Setup

- Running llm version 0.19.1 on Mac OS Sequoia

- Installed the llm plugin `llm-bedrock`

- Made sure I installed the latest `aws` CLI and had the an AWS profile configured with appropriate credentials. I confirmed that it had _Admin_ privileges over everything in that AWS account. 

- Made sure that the Amazon Nova Models were enabled for this AWS account (see below)
<img width="842" alt="image" src="https://github.com/user-attachments/assets/5adb409f-9c03-4e3f-b5e4-aa9b4b4f557d" />

- Made sure that the AWS user has the policy that was defined at the very end of [this article](https://ndurner.github.io/amazon-nova)

- Invoked `llm -m nova-pro` command as follows:
`llm -m nova-pro 'a happy poem about a pelican with a secret'`

However, it returned an error (see below), and the same error was repeated no matter which of the Amazon Nova models I picked

```
Error: An error occurred (AccessDeniedException) when calling the ConverseStream operation: You don't have access to the model with the specified model ID.
```

- Ran the `aws bedrock list-foundation-models | jq '.modelSummaries[] | { id: .modelId, name: .modelName}'` to get the Model IDs. 
<img width="1422" alt="image" src="https://github.com/user-attachments/assets/12678f78-25ab-44fd-a44e-63710e0cbda3" />

However, when I run the `llm models list` command, I am getting this excerpt for Amazon Nova models:
```
Bedrock: us.amazon.nova-micro-v1:0 (aliases: nova-micro)
Bedrock: us.amazon.nova-lite-v1:0 (aliases: nova-lite)
Bedrock: us.amazon.nova-pro-v1:0 (aliases: nova-pro)
```

These have a `us.amazon...` that the actual `aws bedrock` command output does not. Could that be the reason for this error that I am getting?

Bottom line, I cannot seem to test invoking Nova models and sure could use some help here.

**Comment by simonw** [comment]
I ran into trouble trying to land the PR in #14 - it broke for my own AWS account:

> ```bash
> llm -m amazon.nova-micro-v1:0 'hi'
> ```
> > `Error: An error occurred (ValidationException) when calling the ConverseStream operation: Invocation of model ID amazon.nova-micro-v1:0 with on-demand throughput isn’t supported. Retry your request with the ID or ARN of an inference profile that contains this model.`

I'm not sure what to do about this.

**Comment by cloudnativeengineer** [comment]
Found a temporary way to use the Nova micro model and sharing for future readers:

```bash
$ export inferenceProfileArn=$(aws bedrock create-inference-profile \
  --inference-profile-name my-nova-micro \
  --model-source copyFrom=arn:aws:bedrock:us-east-1::foundation-model/amazon.nova-micro-v1:0 \
  --query 'inferenceProfileArn' --output text)
$ git clone https://github.com/simonw/llm-bedrock.git
$ cd llm-bedrock
$ git reset 89c6b38 --hard
HEAD is now at 89c6b38 Docs for nova-premier, refs #18
$ export AWS_REGION=us-east-1
$ sed -i "s|us\.amazon\.nova-micro-v1:0|${inferenceProfileArn}|g" llm_bedrock.py
$ grep application-inference-profile llm_bedrock.py
    ("arn:aws:bedrock:us-east-1:[redacted]:application-inference-profile/eroyqjpemykj", "nova-micro", False),
$ llm install -e .
$ llm models list |grep application-inference-profile
Bedrock: arn:aws:bedrock:us-east-1:[redacted]:application-inference-profile/eroyqjpemykj (aliases: nova-micro)
$ llm -m nova-micro "hello"
Hello! How can I assist you today? Whether you have a question, need information, or just want to chat, I'm here to help
```
